components/admin_console/permission_schemes_settings/permissions_scheme_summary/permissions_scheme_summary.jsx

```diff
@@ -0,0 +1,212 @@
+                    <FormattedMessage
+                        id='admin.permissions.permissionsSchemeSummary.deleting'
+                        defaultMessage='Deleting...'
+                    />
+                </span>
+            );
+        }
+
+        return (
+            <ConfirmModal
+                show={this.state.showConfirmModal}
+                title={title}
+                message={message}
+                confirmButtonText={confirmButton}
+                onConfirm={this.handleDeleteConfirmed}
+                onCancel={this.handleDeleteCanceled}
+            />
+        );
+    }
+
+    stopPropagation = (e) => {
+        e.stopPropagation();
+    }
+
+    handleDeleteCanceled = () => {
+        this.setState({
+            showConfirmModal: false,
+        });
+    }
+
+    handleDeleteConfirmed = async () => {
+        this.setState({deleting: true, serverError: null});
+        const data = await this.props.actions.deleteScheme(this.props.scheme.id);
+        if (data.error) {
+            this.setState({deleting: false, serverError: data.error.message});
+        } else {
+            this.setState({deleting: false, showConfirmModal: false});
+        }
+    }
+
+    delete = (e) => {
+        e.stopPropagation();
+        this.setState({showConfirmModal: true, serverError: null});
+    }
+
+    goToEdit = () => {
+        browserHistory.push('/admin_console/permissions/team-override-scheme/' + this.props.scheme.id);
+    }
+
+    render = () => {
+        const scheme = this.props.scheme;
+
+        let teams = this.props.teams ? this.props.teams.map((team) => (
+            <span
+                className='team'
+                key={team.id}
+            >
+                {team.name}
+            </span>
+        )) : [];
+
+        let extraTeams = null;
+        if (teams.length > 8) {
+            extraTeams = (
+                <OverlayTrigger
+                    trigger={['hover', 'focus']}
+                    delayShow={Constants.OVERLAY_TIME_DELAY}
+                    placement='bottom'
+                    overlay={
+                        <Tooltip
+                            className='team-scheme-extra-teams-overlay'
+                            id={scheme.id + '-extra-teams-overlay'}
+                        >
+                            {teams.slice(8)}
+                        </Tooltip>
+                    }
+                >
+                    <span
+                        className='team'
+                        key='extra-teams'
+                    >
+                        <FormattedMessage
+                            id='admin.permissions.permissionsSchemeSummary.moreTeams'
+                            defaultMessage='+{number} more'
+                            values={{number: teams.length - 8}}
+                        />
+                    </span>
+                </OverlayTrigger>
+            );
+            teams = teams.slice(0, 8);
```

Comment:
What's the number 8 throughout this block and how was it determined?

Reply:
Good question, is the limit of Teams showed in the list before it shows "+X more", and was defined by UX. Said that, makes a lot of sense to move that "Magic Number" to a constant to give it a clear name :)

Reply:
Done! :)